src/test/java/com/google/devtools/build/lib/rules/platform/PlatformInfoApiTest.java

```diff
@@ -0,0 +1,130 @@
+package com.google.devtools.build.lib.rules.platform;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Multimap;
+import com.google.devtools.build.lib.analysis.ConfiguredTarget;
+import com.google.devtools.build.lib.analysis.platform.ConstraintSettingInfo;
+import com.google.devtools.build.lib.analysis.platform.ConstraintValueInfo;
+import com.google.devtools.build.lib.analysis.platform.PlatformProviderUtils;
+import com.google.devtools.build.lib.analysis.util.BuildViewTestCase;
+import com.google.devtools.build.lib.skylarkbuildapi.platform.PlatformInfoApi;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests Skylark API for {@link PlatformInfoApi} providers. */
+@RunWith(JUnit4.class)
+public class PlatformInfoApiTest extends BuildViewTestCase {
+
+  @Test
+  public void testPlatform() throws Exception {
+    platformBuilder().addConstraint("basic", "value1").build();
+    assertNoEvents();
+
+    PlatformInfoApi platformInfo = fetchPlatformInfo();
+    assertThat(platformInfo).isNotNull();
+    ConstraintSettingInfo constraintSetting =
+        ConstraintSettingInfo.create(makeLabel("//foo:basic"));
+    ConstraintValueInfo constraintValue =
+        ConstraintValueInfo.create(constraintSetting, makeLabel("//foo:value1"));
+    assertThat(platformInfo.constraints().get(constraintSetting)).isEqualTo(constraintValue);
+    assertThat(platformInfo.remoteExecutionProperties()).isEmpty();
+  }
+
+  @Test
+  public void testPlatform_overlappingConstraintValueError() throws Exception {
+    checkError(
+        "foo",
+        "my_platform",
+        "Duplicate constraint_values detected: "
+            + "constraint_setting //foo:basic has "
+            + "[//foo:value1, //foo:value2]",
+        "constraint_setting(name = 'basic')",
+        "constraint_value(name = 'value1',",
+        "    constraint_setting = ':basic',",
+        "    )",
```

Comment:
Nit: Spacing is a little awkward. Could you remove the spaces before the close-paren here and below?

Reply:
Done.